### PR TITLE
Add delete resource route to swagger spec

### PR DESCRIPTION
**Pull request by aaron-collier**
Adds the delete path to the swagger spec for resource. As we haven't discussed delete processing yet, this doesn't include any additional processing stub code, just getting the route created.

### CODE REVIEW

File: swagger.json [hunk 1]
```diff
@@ -126,6 +126,36 @@
             "description": "This resource could be updated at this time by TACO."
           }
         }
+      },
+      "delete": {
+        "summary": "Delete a TACO Resource.",
+        "description": "Deletes a TACO resource (Collection, Digital Repository Object, File metadata object [not binary] or subclass of those).",
```
Comment (cmh2166) [inline swagger.json]: not a binary?

File: swagger.json [hunk 2]
```diff
@@ -126,6 +126,36 @@
...
+            }
+          },
+          "400": {
+            "description": "Invalid ID supplied"
+          },
+          "401": {
+            "description": "You are not authorized to delete a resource in TACO."
+          },
+          "500": {
+            "description": "This resource could be updated at this time by TACO."
```
Comment (cmh2166) [inline swagger.json]: "This resource could not be deleted at this time by TACO" or "Your request could not be processed at this time."

File: swagger.json (hunk 1 above)
Comment (cmh2166) [reply swagger.json]: This should explicit delete both the File resource (metadata) *and* File binary, if delete resource identifier points to a File. We do not want the case where we delete the File metadata resource but the binary hangs out with no TACO management pointers existing.

Comment (jcoyne) [reply swagger.json]: @aaron-collier can you add this to the description?
> ... deletes both the File resource (metadata) and File binary, if delete resource identifier points to a File

Comment (aaron-collier) [reply swagger.json]: Adding now, missed that...

**Comment by aaron-collier** [comment]
@cmh2166 would it be more appropriate to rename this to `purgeResource`?

**Comment by cmh2166** [comment]
lets leave it `delete` for now. i asked @hannahfrost yesterday about this, and she said delete was fine at this level of the stack. we can change if it gets confusing.

**Pull request merged**